src/panels/lovelace/components/hui-input-list-editor.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  html,
+  LitElement,
+  PropertyDeclarations,
+  TemplateResult,
+} from "lit-element";
+import "@polymer/paper-input/paper-input";
+
+import { HomeAssistant } from "../../../types";
+import { fireEvent } from "../../../common/dom/fire_event";
+
+import { EditorTarget } from "../editor/types";
+
+export class HuiInputListEditor extends LitElement {
+  protected value?: string[];
+  protected hass?: HomeAssistant;
+  protected heading?: string;
+  protected inputLabel?: string;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      value: {},
+      heading: {},
+      inputLabel: {},
+    };
+  }
+
+  protected render(): TemplateResult | void {
+    if (!this.value) {
+      return html``;
+    }
+
+    return html`
+      ${this.renderStyle()}
+      <h3>${this.heading}</h3>
+      <div class="entries">
+        ${this.value.map((listEntry, index) => {
+          return html`
+            <paper-input
+              label="${this.inputLabel}"
+              .value="${listEntry}"
+              .configValue="${"entry"}"
+              .index="${index}"
+              @value-changed="${this._valueChanged}"
+            ></paper-input>
+          `;
+        })}
+        <paper-input
+          label="${this.inputLabel}"
+          @change="${this._addEntity}"
+        ></paper-input>
+      </div>
+    `;
+  }
+
+  private _addEntity(ev: Event): void {
+    const target = ev.target! as EditorTarget;
+    if (target.value === "") {
+      return;
+    }
+    const newEntries = this.value!.concat(target.value as string);
+    target.value = "";
+    this.value = newEntries;
+    fireEvent(this, "value-changed");
+    (ev.target! as LitElement).blur();
+  }
+
+  private _valueChanged(ev: Event): void {
```

Comment:
you should stop propagation on this event.

Reply:
OK